### PR TITLE
Update pfsense.md

### PR DESCRIPTION
**Pull request by nmindz**
Upstream FreeBSD repository packages have been updated, some packages previous indicated versions no longer work and throw the error `pkg: Not Found`

##### Summary
Update to Documentation

##### Component Name
Agent > Packaging > Installer > Methods > pfSense

##### Test Plan
https://learn.netdata.cloud/docs/agent/packaging/installer/methods/pfsense

##### Additional Information
Updated according to http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/ @ 2020-07-07, 4:45 PM UTC

### CODE REVIEW

**Comment by nmindz** [comment]
Trying to actually run Netdata on pfSense/BSD actually gave me even further errors. Just now I found that I'd also need to 
```
pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/json-c-0.14.txz
```
Which seems to be missing from a standard pfSense install, currently running 2.4.4 (I know there's a patch update available, but that is my production environment.).

**Comment by thiagoftsm** [comment]
> Trying to actually run Netdata on pfSense/BSD gave me even further errors. Just now I found that I'd also need to
> 
> ```
> pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/json-c-0.14.txz
> ```
> 
> Which seems to be missing from a standard pfSense install, currently running 2.4.4 (I know there's a patch update available, but that is my production environment.).

Hi @nmindz,

Thank you for your contribution!
Good catch, please, can you also add this link in our documentation you are changing? Could you also sign the [CLA](https://github.com/netdata/netdata/pull/9499#issuecomment-654988574) for we merge your contribution?

Best regards!

**Comment by nmindz** [comment]
Hello @thiagoftsm !

I had already signed the CLA. 😃 

I also added the extra missing package as requested, and an additional warning/disclaimer right after the package list and the other 2 notices, regarding the `latest` tag from the FreeBSD 11 repository and implications for people running older (`11.2-RELEASE` and older) versions.

**Comment by thiagoftsm** [comment]
I verified that I can do download of all packages on my host and when I installed Netdata on PfSense 2.4.5 the packages was installed with success, but for Netdata runs, I had to create the next symbolic link

```
ln -s /usr/local/lib/libjson-c.so.4.0.0 /usr/local/lib/libjson-c.so.5
```

@nmindz did you also had to run the previous command? If you had, please, can you also add it to the documentation.

**Comment by prologic** [comment]
@nmindz ping? Please help @thiagoftsm :)

cc @joelhans Have you had a change to review this doc update yet?

Content here looks good technically, next logical steps we should take should be to codify this up into `install-required-packages.sh` as we are starting to support *BSD more officially now.

**Comment by nmindz** [comment]
# @thiagoftsm Nope, I didn't. But, in fact, I'm running pfSense 2.4.4 so that would be equivalent to `11.2-RELEASE`.

That might essentially differ and somehow not require a symlink to that shared object.

However, on a side note, I didn't actually get to run Netdata (other than simply running `service netdata onestart` and killing it shortly after) because I intended to run it along with the cloud agent, however the package has been built with `--disable-cloud` which is a bummer.

Since this is a production server, I didn't quite have the opportunity to venture into using jails on it (I'd rather not mess too much with the appliance instance), but I have neither ever run jails on pfSense, let alone on a slightly outdated version still in critical daily. usage.

I might try it later at some sandbox but currently I'm on a tight schedule and can't commit to a date on that sandboxed test.

**Comment by thiagoftsm** [comment]
Hello @nmindz,

I agree with @prologic the content is ok, when you have a time, please, take a look in this missing symbolic link, I remember in the past I also had to do this with the same library, unfortunately this happens sometimes, because we are using FreeBSD package.

Best regards!

File: packaging/installer/methods/pfsense.md [hunk 1]
```diff
@@ -16,26 +16,30 @@ Netdata, Judy and Python are downloaded from the FreeBSD repository.
...
 
 **Note:** On pfSense 2.4.5, Python version 3.7 may be installed by the system, in which case you should should not install Python from the FreeBSD repository as instructed above.
 
+**WARNING:** Packages from the `latest` tag in the FreeBSD 11 repository will always be at the latest stable train. As of early July 2020, that means 11.3-RELEASE. If you are running earlier versions of pfSense, such as 2.4.4 (now outdated, but might be running in some edge cases in production environments) at this time, then you'll be running __11.2-RELEASE__. Be mindful of your update train or you might cause havoc in your system! **BE MINDFUL OF YOUR PACKAGE MANAGER WARNINGS, THEY SHOULD NOT BE DEALT WITH LIGHTLY!** - With that said, if you are running an older version, you might need to manually browse for the correct package versions for your release train. For ease and reference, [11.3-RELEASE](http://pkg.freebsd.org/FreeBSD:11:amd64/release_3/), [11.2-RELEASE](http://pkg.freebsd.org/FreeBSD:11:amd64/release_2/), [11.1-RELEASE](http://pkg.freebsd.org/FreeBSD:11:amd64/release_1/) and [11.0-RELEASE](http://pkg.freebsd.org/FreeBSD:11:amd64/release_0/) are linked here.
```
Comment (joelhans) [inline packaging/installer/methods/pfsense.md:41]:
```suggestion
**WARNING:** Packages from the `latest` tag in the FreeBSD 11 repository will always be at the latest stable train. As of early July 2020, that means `11.3-RELEASE`. If you are running earlier versions of pfSense, such as 2.4.4 at this time, then you'll be running `11.2-RELEASE`. Be mindful of your update train, and pay close attention to warnings from your package manager, as they should not be dealt with lightly.

if you are running an older version, you might need to manually browse for the correct package versions for your release train: [11.3-RELEASE](http://pkg.freebsd.org/FreeBSD:11:amd64/release_3/), [11.2-RELEASE](http://pkg.freebsd.org/FreeBSD:11:amd64/release_2/), [11.1-RELEASE](http://pkg.freebsd.org/FreeBSD:11:amd64/release_1/), and [11.0-RELEASE](http://pkg.freebsd.org/FreeBSD:11:amd64/release_0/).
```

File: packaging/installer/methods/pfsense.md [hunk 2]
```diff
@@ -16,26 +16,32 @@ Netdata, Judy and Python are downloaded from the FreeBSD repository.
...
 pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-six-1.14.0.txz
 pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-cryptography-2.6.1.txz
 pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-idna-2.8.txz
 pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-openssl-19.0.0.txz
 pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-pysocks-1.7.1.txz
 pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-urllib3-1.25.7,1.txz
-pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-yaml-5.2.txz
-pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/netdata-1.20.0_3.txz
+pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-yaml-5.3.1.txz
+pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/netdata-1.23.0.txz
```
Comment (ilyam8) [inline packaging/installer/methods/pfsense.md:31]: good catch @joelhans 

```cmd
0 ~ $ curl -o /dev/null -s -w "%{http_code}\n" http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/netdata-1.23.0.txz
404
0 ~ $ curl -o /dev/null -s -w "%{http_code}\n" http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/netdata-1.24.0.txz
200
```

Comment (ilyam8) [inline packaging/installer/methods/pfsense.md:31]:
```suggestion
pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/netdata-1.26.0.txz
```

```cmd
0 ~ $ curl -o /dev/null -s -w "%{http_code}\n" http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/netdata-1.26.0.txz
200
```

**Comment by OdysLam** [comment]
Hey everyone,

There is a user who ask about updating his pfsense Netdata installation. He simply needs to re-run 
`pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/netdata-1.2X.0.txz` with the latest version of netdata in place of the X ? 
cc @ilyam8 @joelhans @thiagoftsm

**Comment by thiagoftsm** [comment]
> There is a user who ask about updating his pfsense Netdata installation. He simply needs to re-run
> `pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/netdata-1.2X.0.txz` with the latest version of netdata in place of the X ?

If there was a previous installation, the user does not need to run `pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/netdata-1.2X.0.txz` again, the user only needs to run `pkg update` when the new package is available.

**Comment by OdysLam** [comment]
I see,  thanks @thiagoftsm for the swift response!

**Comment by ilyam8** [comment]
i checked other packages

```bash
PACKAGES=(
  http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/Judy-1.0.5_2.txz
  http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/json-c-0.14.txz
  http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-certifi-2020.6.20.txz
  http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-asn1crypto-1.3.0.txz
  http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-pycparser-2.20.txz
  http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-cffi-1.14.0.txz
  http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-six-1.14.0.txz
  http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-cryptography-2.6.1.txz
  http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-idna-2.8.txz
  http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-openssl-19.0.0.txz
  http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-pysocks-1.7.1.txz
  http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-urllib3-1.25.7,1.txz
  http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-yaml-5.3.1.txz
  http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/netdata-1.26.0.txz
)

for pkg in ${PACKAGES[@]}; do echo "$pkg"; curl -o /dev/null -s -w "%{http_code}\n" "$pkg"; done
```

```cmd
0 ~ $ for pkg in ${PACKAGES[@]}; do echo "$pkg"; curl -o /dev/null -s -w "%{http_code}\n" "$pkg"; done
http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/Judy-1.0.5_2.txz
200
http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/json-c-0.14.txz
404
http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-certifi-2020.6.20.txz
200
http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-asn1crypto-1.3.0.txz
200
http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-pycparser-2.20.txz
200
http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-cffi-1.14.0.txz
404
http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-six-1.14.0.txz
404
http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-cryptography-2.6.1.txz
200
http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-idna-2.8.txz
404
http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-openssl-19.0.0.txz
200
http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-pysocks-1.7.1.txz
200
http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-urllib3-1.25.7,1.txz
404
http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-yaml-5.3.1.txz
200
http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/netdata-1.26.0.txz
200
```

File: packaging/installer/methods/pfsense.md [hunk 3]
```diff
@@ -16,26 +16,32 @@ Netdata, Judy and Python are downloaded from the FreeBSD repository.
 ```sh
 pkg install -y pkgconf bash e2fsprogs-libuuid libuv nano
 pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/Judy-1.0.5_2.txz
-pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-certifi-2020.4.5.1.txz
+pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/json-c-0.14.txz
```
Comment (ilyam8) [inline packaging/installer/methods/pfsense.md:19]:
```suggestion
pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/json-c-0.15_1.txz
```

File: packaging/installer/methods/pfsense.md [hunk 4]
```diff
@@ -16,26 +16,32 @@ Netdata, Judy and Python are downloaded from the FreeBSD repository.
...
+pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-certifi-2020.6.20.txz
 pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-asn1crypto-1.3.0.txz
-pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-pycparser-2.19.txz
+pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-pycparser-2.20.txz
 pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-cffi-1.14.0.txz
```
Comment (ilyam8) [inline packaging/installer/methods/pfsense.md:23]:
```suggestion
pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-cffi-1.14.3.txz
```

File: packaging/installer/methods/pfsense.md [hunk 5]
```diff
@@ -16,26 +16,32 @@ Netdata, Judy and Python are downloaded from the FreeBSD repository.
...
 pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-six-1.14.0.txz
```
Comment (ilyam8) [inline packaging/installer/methods/pfsense.md:24]:
```suggestion
pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-six-1.15.0.txz
```

File: packaging/installer/methods/pfsense.md [hunk 6]
```diff
@@ -16,26 +16,32 @@ Netdata, Judy and Python are downloaded from the FreeBSD repository.
...
 pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-cryptography-2.6.1.txz
 pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-idna-2.8.txz
```
Comment (ilyam8) [inline packaging/installer/methods/pfsense.md:26]:
```suggestion
pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-idna-2.10.txz
```

File: packaging/installer/methods/pfsense.md [hunk 7]
```diff
@@ -16,26 +16,32 @@ Netdata, Judy and Python are downloaded from the FreeBSD repository.
...
 pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-openssl-19.0.0.txz
 pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-pysocks-1.7.1.txz
 pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-urllib3-1.25.7,1.txz
```
Comment (ilyam8) [inline packaging/installer/methods/pfsense.md:29]:
```suggestion
pkg add http://pkg.freebsd.org/FreeBSD:11:amd64/latest/All/py37-urllib3-1.25.10,1.txz
```

**Comment by Ferroin** [comment]
This is going to continue to be a moving target. Perhaps instead of listing the exact URLs we should tell users how to find the correct URLs for the required packages so that they get the most recent versions?

**Comment by ilyam8** [comment]
Yeah, it doesn't work as it is

**Comment by joelhans** [comment]
@Ferroin Can you think of any better method than using the [FreeBSD ports search](https://www.freebsd.org/ports/) for each package name and finding the latest version number? That's all I can come up with. If so, the doc will list package names and tell users they need to search for the latest versions.

It's not a good experience, but either is a bunch of always-outdated package versions...

**Comment by nmindz** [comment]
@joelhans I think the second best option would be for Netdata to provide a static mirror to packages tested/compatible with a given release version and have those links maintained in the docs until they require further update. That way you remove the user's responsibility of searching for packages (and therefore picking a wrong/malfunctioning one) and you have to update the doc less often (no need to try and compete with FreeBSD's packages release cycles).

**Comment by Ferroin** [comment]
Honestly, the ideal situation IMO would be to have a script that figures out the required packages and installs them. Long-term we kind of need that anyway if we want the dependency handling to work reliably on FreeBSD for our official install scripts.

**Comment by ilyam8** [comment]
Closed because the PR got outdated

**Pull request closed without merging**